data/changelog/dune/2024-01-16-dune-3.13.0.md

```diff
@@ -0,0 +1,128 @@
+    its own sublibraries (#9461, fixes #9456, @jonludlam)
+  
+  - Resolve various public binaries to their build location, rather than to where
+    they're copied in the `_build/install` directory (#9496, fixes #7908,
+    @rgrinberg).
+  
+  - Correctly ignore warning flags in vendored projects (#9515, @rgrinberg)
+  
+  - Use watch exclusions in watch mode on MacOS (#9643, fixes #9517,
+    @PoorlyDefinedBehaviour)
+  
+  - Fix merlin configuration for `(include_subdirs qualified)` modules (#9659,
+    fixes #8297, @rgrinberg)
+  
+  - Fix handling of `enabled_if` in binary install stanzas. Previously, we'd
+    ignore the result of `enabled_if` when evaluating `%{bin:..}` (#9707,
+    @rgrinberg)
+---
+
+We're happy to announce that Dune 3.13.0 is now available.
+This feature is packed with fixes and new features that you can find in the
+changelog.
+
+There are a few new features that we would like to specially highlight.
+
+### Generate Conflicts File for Menhir Grammars ([#9512](https://github.com/ocaml/dune/pull/9512), [@nojb](https://github.com/nojb))
+
+When `menhir` is used to generate code from `.mly` files, there are sometimes
+issues with the grammar itself, such as shift-reduce conflicts.
+
+Menhir has an option to generate a "conflicts" file using its `--explain` flag,
+but until now this was not exposed by Dune. Starting from this version, this
+file will be generated automatically to help developers debug their grammars.
+
+### Cached Directory Targets ([#9535](https://github.com/ocaml/dune/pull/9535), [@rleshchinskiy](https://github.com/rleshchinskiy))
+
+Dune's global cache is a way to save the result of intermediate build results,
+even across projects. Previously, it would only work with file targets. With
+this change, it now supports Dune's experimental directory targets.
+
+### Dynamic Module List ([#9578](https://github.com/ocaml/dune/pull/9578), [@nojb](https://github.com/nojb))
+
+In several places in the Dune language, it is possible to pass a list of
+modules. For example in a `(library)` stanza, if for some reason the default of
+picking all the source files in the current directory is not the right thing to
+do, it is possible to pass `(module A B C)` to only attach these modules to the
+library.
+
+An important limitation has been that the list of modules needed to be static:
+written as is in the `dune` file. This limitation has now been lifted and it is
+possible to use `(:include)` or `%{read-lines:file}` in this field and similar
+ones.
+
+Lexifi's use case is a system of static plug-ins: a program is extended by
+selecting which modules are linked to the application core. This list of
+modules can now be emitted by a generator that reads a configuration file.
+
+Previously, this required using OCaml syntax for the `dune` file, which has
+poor performance because Dune does not know the dependencies of the generator
+and needs to re-run the build more times than necessary.
```

Comment:
Also, `dune` files in OCam syntax have a number of limitations (eg they cannot be used in combination with `(include_subdirs)`.